src/lib/table/table.md

```diff
@@ -1,50 +1,185 @@
-sort headers emit events that can be used to trigger an update via the table's data source.
+To add sorting behavior to the table, add the `matSort` directive to the `<mat-table>` and add 
+`mat-sort-header` to each column header cell that should trigger sorting. Provide the `MatSort` 
+directive to the `MatTableDataSource` and it will automatically listen for sorting changes and 
+change the order of data rendered by the table.
+
+By default, the `MatTableDataSource` sorts with the assumption that the sorted column's name 
+matches the data property name that the column displays. For example, the following column 
+definition is named `position`, which matches the name of the property displayed in the row cell.
+
+```html
+<!-- Name Column -->
+<ng-container matColumnDef="position">
+  <mat-header-cell *matHeaderCellDef mat-sort-header> Name </mat-header-cell>
+  <mat-cell *matCellDef="let element"> {{element.position}} </mat-cell>
+</ng-container>
+```
+
+If the data properties do not match the column names, or if a more complex data property accessor 
+is required, then a custom `sortingDataAccessor` function can be set to override the default data 
+accessor on the `MatTableDataSource`.
 
 <!-- example(table-sorting) -->
 
+For more information on using and configuring the sorting behavior, check out the 
+[matSort docs](https://material.angular.io/components/sort/overview).
+
 #### Filtering
 
-While Angular Material does not offer a specific component for filtering tabular data, the table's
-data source can be updated based on any custom filter UI. Any filtering pattern need only trigger
-an update via the table's data source.
+To remove filtered rows from the table's data, simply provide the filter string to the 
+`MatTableDataSource`. The data source will reduce each row data to a serialized form and will 
+filter out the row if it does not contain the filter string. By default, the row data reducing 
+function will concatenate all the object values and convert them to lowercase.
 
-<!--- example(table-filtering) -->
+For example, the data object `{id: 123, name: 'Mr. Smith', favoriteColor: 'blue'}` will be 
+reduced to `123mr. smithblue`. If your filter string was `blue` then it would be considered a match 
+because it is contained in the reduced string, and the row would be displayed in the table.
 
-### Simple Table
+To override the default filtering behavior, a custom `filterPredicate` function can be set 
+which takes a data object and filter string and returns true if the data object is considered a 
+match.
+
+<!--- example(table-filtering) -->
 
-In the near future, we will provide a simplified version of the data-table with an easy-to-use
-interface, material styling, array input, and more out-of-the-box features (sorting, pagination,
-and selection).
+#### Selection
+
+To add row selection to the table, first set up a `SelectionModel` from `@angular/cdk/collections` 
```

Comment:
Same comment here about specificity- I want to keep it clear that users can do selection however they want. If we present it as "this is the way you are supposed to do selection", it begs the question as to why it's not just built in. I would really rather include this in an example with explanation that say "Here is the One True Way to do selection" (same for filtering)

Reply:
Flexibility is nice, but most developers are just going to want a working example they can start with.